source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -371,6 +371,24 @@ const std::string& SslSocketInfo::sha256PeerCertificateDigest() const {
   return cached_sha_256_peer_certificate_digest_;
 }
 
+const std::string& SslSocketInfo::sha1PeerCertificateDigest() const {
+  if (!cached_sha_1_peer_certificate_digest_.empty()) {
+    return cached_sha_1_peer_certificate_digest_;
+  }
+  bssl::UniquePtr<X509> cert(SSL_get_peer_certificate(ssl()));
+  if (!cert) {
+    ASSERT(cached_sha_1_peer_certificate_digest_.empty());
+    return cached_sha_1_peer_certificate_digest_;
+  }
+
+  std::vector<uint8_t> computed_hash(SHA_DIGEST_LENGTH);
```

Comment:
@lizan Not fully familiar with open ssl code - Is using `SHA_DIGEST_LENGTH` for `SHA1` digest correct?

Reply:
Yes I think that's right.